### ISSUE TITLE
grp is not patternable & causes a crash

### ISSUE THREAD

**Issue by jarmitage** [tidalcycles/Tidal#639]
`<>` `[]` etc don't work with `grp` e.g

```
del = grp [mF "delay", mF "delaytime", mF "delayfeedback"]
```
...
```
d1 $ s "bd" # del "0.5:<0.5>:0.5"
```
`Syntax error in sequence:  "0.5:<0.5>:0.5"`

**Comment by bgold-cosmos** [comment]
I don't have a solution, but in case it's useful this happens because `:` is interpreted at the "event" level rather than working with subpatterns on each side.  E.g. a pattern like `s "foo:1 bar:2"` is considered a pattern of two events, "foo:1" and "bar:2", and only after that is it broken up.

Having `:` work on subpatterns is probably possible, but it would have to deal with all the pattern structure questions that led to all the different merging operators ('|>', '>|', ...).  For example, `s "[foo bar]:[1 2 3]"` could be equivalent to `s "foo:1 bar:2"` or `s "foo:1 foo:2 bar:3"` or `s "foo:1 [foo:2 bar:2] bar:3"`

**Comment by jarmitage** [comment]
Thanks @bgold-cosmos.

Maybe it’s better to leave `grp` as is then, or change its syntax to something other than `:`, although I’m not sure what to.

I have `grp'` functions defined too for when I want to use functions on a parameter group, which also works with patterns. Only problem there is you can’t take advantage of structure operators.

**Comment by yaxu** [comment]
Duplicate of #629.
